src/AWS.Deploy.Orchestration/Orchestrator.cs

```diff
@@ -238,6 +235,15 @@ public async Task ApplyAllReplacementTokens(Recommendation recommendation, strin
                 var vpcs = await _awsResourceQueryer.GetListOfVpcs();
                 recommendation.AddReplacementToken(Constants.RecipeIdentifier.REPLACE_TOKEN_HAS_NOT_VPCS, !vpcs.Any());
             }
+            if (recommendation.ReplacementTokens.ContainsKey(Constants.RecipeIdentifier.REPLACE_TOKEN_DEFAULT_CONTAINER_PORT))
+            {
+                if (_dockerEngine == null)
+                    throw new InvalidOperationException($"{nameof(_dockerEngine)} is null as part of the Orchestrator object");
+
+                var defaultPort = _dockerEngine.DetermineDefaultDockerPort(recommendation);
+                recommendation.AddReplacementToken(Constants.RecipeIdentifier.REPLACE_TOKEN_DEFAULT_CONTAINER_PORT, defaultPort);
+                recommendation.DeploymentBundle.DockerfileHttpPort = defaultPort;
```

Comment:
Mostly for my understanding: is setting this on the `DeploymentBundle` necessary? I don't see us doing it for other members with replacement tokens.

Reply:
It is necessary since we usually set the deployment bundle settings whenever a setting is updated. When we apply replacements and directly deploy with no setting modification, the deployment bundle does not get updated to take in the replacement. The other settings that are in the same boat as this new setting, all set the deployment bundle settings at different parts of the process. We don't do it in a standardized way.